### PR TITLE
SALTO-6129: Fix tests build

### PR DESCRIPTION
**Pull request by shir-reifenberg**
Removed the additional tsconfig defined for tests and fixed all build errors

Thanks @tamtamirr for helping with the SF errors!

---

_Additional context for reviewer_

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/salesforce-adapter/tsconfig.json [hunk 1]
```diff
@@ -3,12 +3,14 @@
...
+    "declaration": true,
+    "resolveJsonModule": true,
     "paths": {
       "@salto-io/jsforce": ["../../node_modules/@salto-io/jsforce-types"]
     }
   },
-  "include": ["src/**/*", "src/**/*.json", "index.ts", "e2e_test/jest_environment.ts"],
-  "exclude": ["test/sfdx_parser/test_sfdx_project/**"],
+  "include": ["src/**/*", "src/**/*.json", "test/**/*", "e2e_test/**/*", "index.ts"],
+  "exclude": ["test/sfdx_parser/test_sfdx_project/**", "test/adapter.crud.test.ts"],
```
Comment (shir-reifenberg) [inline packages/salesforce-adapter/tsconfig.json:13]: temp until I'll be able to resolve the build errors in this file

File: packages/okta-adapter/e2e_test/jest_matchers.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      pass,
+      message: () =>
+        `Received unexpected result when fetching instance: ${expected.elemID.getFullName()}.\n` +
+        `Expected value: ${inspectValue(expected.value, { depth: 7 })},\n` +
+        `Received value: ${inspectValue(received, { depth: 7 })}`,
+    }
+  },
+})
+
+interface CustomMatchers<R = unknown> {
```
Comment (Nurdok) [inline packages/okta-adapter/e2e_test/jest_matchers.ts:25]: Nice.

File: packages/persistent-pool/tsconfig.json [hunk 3]
```diff
@@ -4,6 +4,6 @@
     "outDir": "dist",
     "baseUrl": "."
   },
-  "include": ["src/**/*", "test/**/*", "index.ts"],
+  "include": ["src/**/*", "test/**/*", "index.ts", "../lowerdash/test/retry/wait_for.test.ts"],
```
Comment (Nurdok) [inline packages/persistent-pool/tsconfig.json:7]: Why does lowerdash needs to be here?

**Review by Nurdok — APPROVED** [review]

File: packages/persistent-pool/tsconfig.json (hunk 3 above)
Comment (shir-reifenberg) [reply packages/persistent-pool/tsconfig.json:7]: shouldn't, good catch!

**Pull request merged**